streams/src/test/java/org/apache/kafka/streams/kstream/internals/KStreamKTableJoinTest.java

```diff
@@ -57,82 +60,182 @@ public void setUp() throws IOException {
+        // push two items to the primary stream. the table is empty
 
         for (int i = 0; i < 2; i++) {
-            driver.process(topic1, expectedKeys[i], "X" + expectedKeys[i]);
+            driver.process(streamTopic, expectedKeys[i], "X" + expectedKeys[i]);
         }
 
         processor.checkAndClearProcessResult();
 
-        // push two items to the other stream. this should not produce any item.
+        // push two items to the table. this should not produce any item.
 
         for (int i = 0; i < 2; i++) {
-            driver.process(topic2, expectedKeys[i], "Y" + expectedKeys[i]);
+            driver.process(tableTopic, expectedKeys[i], "Y" + expectedKeys[i]);
         }
 
         processor.checkAndClearProcessResult();
 
         // push all four items to the primary stream. this should produce two items.
 
-        for (int expectedKey : expectedKeys) {
-            driver.process(topic1, expectedKey, "X" + expectedKey);
+        for (final int expectedKey : expectedKeys) {
+            driver.process(streamTopic, expectedKey, "X" + expectedKey);
         }
 
         processor.checkAndClearProcessResult("0:X0+Y0", "1:X1+Y1");
 
-        // push all items to the other stream. this should not produce any item
-        for (int expectedKey : expectedKeys) {
-            driver.process(topic2, expectedKey, "YY" + expectedKey);
+        // push all items to the table. this should not produce any item
+        for (final int expectedKey : expectedKeys) {
+            driver.process(tableTopic, expectedKey, "YY" + expectedKey);
         }
 
         processor.checkAndClearProcessResult();
 
         // push all four items to the primary stream. this should produce four items.
 
-        for (int expectedKey : expectedKeys) {
-            driver.process(topic1, expectedKey, "X" + expectedKey);
+        for (final int expectedKey : expectedKeys) {
+            driver.process(streamTopic, expectedKey, "X" + expectedKey);
         }
 
         processor.checkAndClearProcessResult("0:X0+YY0", "1:X1+YY1", "2:X2+YY2", "3:X3+YY3");
 
-        // push two items with null to the other stream as deletes. this should not produce any item.
+        // push two items with null to the table as deletes. this should not produce any item.
 
         for (int i = 0; i < 2; i++) {
-            driver.process(topic2, expectedKeys[i], null);
+            driver.process(tableTopic, expectedKeys[i], null);
         }
 
         processor.checkAndClearProcessResult();
 
         // push all four items to the primary stream. this should produce two items.
 
-        for (int expectedKey : expectedKeys) {
-            driver.process(topic1, expectedKey, "XX" + expectedKey);
+        for (final int expectedKey : expectedKeys) {
+            driver.process(streamTopic, expectedKey, "XX" + expectedKey);
         }
 
         processor.checkAndClearProcessResult("2:XX2+YY2", "3:XX3+YY3");
     }
+
+
+    @Test
+    public void testGlobalKTableJoin() {
```

Comment:
I am must wondering, if this should go into it's own test class `KStreamGlobalKTableJoinTest.java` ? Similar below.

Reply:
Yeah - I was thinking the same thing. I've pushed an update with the globalktable tests split out into separate classes.
I also split them out into more fine-grained test methods as per @bbejeck 's suggestion